camel/agents/task_agent.py

```diff
@@ -151,3 +154,75 @@ def step(
 
         sub_tasks_msg = sub_tasks_msgs[0]
         return TextPrompt(sub_tasks_msg.content)
+
+
+class RoleAssignmentAgent(ChatAgent):
+    r"""
+    An agent that generates role names based on the task prompt.
+    Attributes:
+        role_assignment_prompt (TextPrompt): A prompt for the agent to generate
+        role names.
+    args:
+        model (ModelType): The tupe of model to use for the agent.
+            (default: :obj: 'ModelType.GPT_3_5_TURBO')
+        model_config (Any): The configuration for the model.
+            (default: :obj: 'None')
+    """
+
+    def __init__(
+        self,
+        model: ModelType = ModelType.GPT_3_5_TURBO,
+        model_config: Any = None,
+    ) -> None:
+        self.role_assignment_prompt = TextPrompt(
+            "Given this task, \"{task}\", generate two role names, one for the"
+            + " AI user and one for the AI assistant.")
+
+        system_message = SystemMessage(
+            role_name="Role Assigner",
+            role_type=RoleType.ASSISTANT,
+            content="You assign roles based on tasks.",
+        )
+
+        super().__init__(system_message, model, model_config)
+
+    def step(
+        self,
+        task_prompt: Union[str, TextPrompt],
+    ) -> Tuple[TextPrompt, TextPrompt]:
+        r""""
+        Generate role names based on the input task prompt.
+        Args:
+            task_prompt (Union[str, TextPrompt]): The prompt
+            for the task based
+            on which the roles are to be generated.
+            Returns:
+                Tuple[TextPrompt, Textprompt]: The role names generated by the
+                agent for
+                the AI user and AI assistant, respectively.
+        """
+
+        self.reset()
+
+        task_msg = AssistantChatMessage(
+            role_name="Role Assigner",
```

Comment:
AssistantChatMessage is long gone. Where did you branch from?

Reply:
I modified based on this version of code: git clone -b v0.1.0 https://github.com/camel-ai/camel.git

Reply:
@lightaime You were right about the possible misinterpretation of this line in the instruction...